signing/src/main/java/tech/pegasys/web3signer/signing/config/SignerLoader.java

```diff
@@ -67,18 +71,37 @@ public Collection<ArtifactSigner> load(
     return processMetadataFilesInParallel(configFilesContent, signerParser);
   }
 
-  private Map<Path, String> getConfigFilesContents(
+  private static void updateLoadedConfigFilesMap() {
+    // remove loaded config files which do not exist anymore
+    new HashSet<>(loadedConfigFiles.keySet())
```

Comment:
Wrapping this in another set shouldn't necessary if we are using a concurrent hashmap

Reply:
wouldn't it cause an issue in `forEach`? as we would be removing an item from same iterator.

Reply:
Yeah it would do